### PR TITLE
[CONSUL-405] Add bats to single container

### PR DESCRIPTION
**Pull request by iberlot**
### Description
The Bats Scripts should be working correctly in the unified container. When running the tests, it has to be executed directly in that container through a Docker exec.

### CODE REVIEW

File: test/integration/connect/envoy/run-tests.windows.sh [hunk 1]
```diff
@@ -332,14 +332,7 @@ function verify {
...
   # 1 because we're using --pid=host so we use TINI_SUBREAPER
-  if docker.exe run --name envoy_verify-${CLUSTER}_1 -t \
-    -e TINI_SUBREAPER=1 \
-    -e ENVOY_VERSION \
-    $WORKDIR_SNIPPET \
-    --pid=host \
-    $(network_snippet $CLUSTER) \
-    bats-verify \
-    --pretty ${CLUSTER}/bats ; then
+  if docker.exe exec -i envoy_consul-${CLUSTER}_1 bash -c "/c/bats/bin/bats /c/workdir/${CLUSTER}/bats/verify.bats" ; then
```
Comment (cocolavayen) [inline test/integration/connect/envoy/run-tests.windows.sh]: Please let's include `--pretty` parameter like the old command. 
Let's also add a line break as the old one

Comment (cocolavayen) [inline test/integration/connect/envoy/run-tests.windows.sh]: Let's use the `SINGLE_CONTAINER_BASE_NAME` variable declared in the `line 12` to be consistent with the others docker exec commands and any change is easier

File: execute_test.sh [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    go test -v -timeout=30m -tags integration ./test/integration/connect/envoy -run="TestEnvoy/case-dogstatsd-udp" -win=true > output/case-dogstatsd-udp.txt
+    go test -v -timeout=30m -tags integration ./test/integration/connect/envoy -run="TestEnvoy/case-grpc" -win=true > output/case-grpc.txt
+    go test -v -timeout=30m -tags integration ./test/integration/connect/envoy -run="TestEnvoy/case-http" -win=true > output/case-http.txt
+    go test -v -timeout=30m -tags integration ./test/integration/connect/envoy -run="TestEnvoy/case-http-badauthz" -win=true > output/case-http-badauthz.txt
+    go test -v -timeout=30m -tags integration ./test/integration/connect/envoy -run="TestEnvoy/case-statsd-udp" -win=true > output/case-statsd-udp.txt
+    go test -v -timeout=30m -tags integration ./test/integration/connect/envoy -run="TestEnvoy/case-stats-proxy" -win=true > output/case-stats-proxy.txt
+    echo "completado 100%"
+fi
+
+echo "Completed test"
```
Comment (joselo85) [inline execute_test.sh]: Add a new line here, and if it is possible use english on the echos (suggestion).

**Review by joselo85 — CHANGES_REQUESTED** [review]

File: test/integration/connect/envoy/helpers.windows.bash [hunk 3]
```diff
@@ -886,7 +822,7 @@ function getIP {
 }
 
 function getIP_container {
-  docker inspect -f '{{range.NetworkSettings.Networks}}{{.IPAddress}}{{end}}' $1
+  docker.exe inspect -f '{{range.NetworkSettings.Networks}}{{.IPAddress}}{{end}}' $1
```
Comment (cocolavayen) [inline test/integration/connect/envoy/helpers.windows.bash:824]: Are we still using this function? and the getIP as well?

File: test/integration/connect/envoy/run-tests.windows.sh [hunk 4]
```diff
@@ -332,14 +332,7 @@ function verify {
...
   # 1 because we're using --pid=host so we use TINI_SUBREAPER
-  if docker.exe run --name envoy_verify-${CLUSTER}_1 -t \
-    -e TINI_SUBREAPER=1 \
-    -e ENVOY_VERSION \
-    $WORKDIR_SNIPPET \
-    --pid=host \
-    $(network_snippet $CLUSTER) \
-    bats-verify \
-    --pretty ${CLUSTER}/bats ; then
+  if docker.exe exec -i ${SINGLE_CONTAINER_BASE_NAME}-${CLUSTER}_1 bash -c "/c/bats/bin/bats --pretty /c/workdir/${CLUSTER}/bats" ; then
```
Comment (cocolavayen) [inline test/integration/connect/envoy/run-tests.windows.sh]: Are not needed the environments variables TINI_SUBREAPER and ENVOY_VERSION?
I noted that we have bats in the Path, so we can use it directly without indicate its path

```suggestion
  if docker.exe exec -t $CONTAINER_NAME bash.exe -c "TINI_SUBREAPER=1 \
                                                      ENVOY_VERSION=${ENVOY_VERSION} \
                                                      bats --pretty /c/workdir/${CLUSTER}/bats"; then
```

File: test/integration/connect/envoy/helpers.windows.bash (hunk 3 above)
Comment (iberlot) [reply test/integration/connect/envoy/helpers.windows.bash:824]: We do not review it, we can add it for the cleaning task

File: build-support-windows/Dockerfile-consul-dev-windows [hunk 5]
```diff
@@ -25,4 +33,4 @@ EXPOSE 19000 19001 19002 19003 19004
 EXPOSE 21000 21001 21002 21003 21004
 EXPOSE 5000 1234 2345
 
-RUN SETX /M path "%PATH%;C:\consul;C:\envoy;C:\fortio;C:\Program Files\Git\bin;C:\Program Files\OpenSSL-Win64\bin"
+RUN SETX /M path "%PATH%;C:\consul;C:\envoy;C:\fortio;C:\Program Files\Git\bin;C:\Program Files\OpenSSL-Win64\bin;C:\bats\bin\bats;C:\ProgramData\chocolatey\lib\jq\tools;"
```
Comment (cocolavayen) [inline build-support-windows/Dockerfile-consul-dev-windows:36]: is the Path to bats binary OK?
I think that it's `C:\bats\bin` instead `C:\bats\bin\bats`

Comment (iberlot) [reply build-support-windows/Dockerfile-consul-dev-windows:36]: The Bats Scripts should be working correctly in the unified container. When running the tests, it has to be executed directly in that container through a Docker exec.

File: test/integration/connect/envoy/helpers.windows.bash [hunk 6]
```diff
@@ -159,7 +103,7 @@ function is_set {
 
 function get_cert {
   check_hostport $1
```
Comment (joselo85) [inline test/integration/connect/envoy/helpers.windows.bash]: Delete check_hostport $1

**Review by joselo85 — CHANGES_REQUESTED** [review]

**Review by ezfepo — APPROVED** [review]

**Pull request merged**